keps/sig-node/2625-cpumanager-policies-thread-placement/README.md

```diff
@@ -0,0 +1,346 @@
+  This guarantees that no physical core is shared among different containers, which improves cache efficiency and mitigates the interference
+  with other workloads that can consume resources of the same physical core, e.g. first level caches.
+
+## Proposal
+
+### User Stories
+
+#### Latency-sensitive applications runtime guarantees
+
+Some classes of latency-sensitive applications (CNF, HFT, ML/AI) benefit of the thread placement constraints this policy enables.
+More precise thread placement allows to control which physical cores are shared among containers, such as:
+
+1. workloads may ensure the physical cores are shared among their threads for increased efficiency
+2. workloads may ensure the physical cores are *not* shared with other containers for interference prevention.
+
+An implementation of the concepts proposed here is already found in [external projects](https://github.com/nokia/CPU-Pooler#hyperthreading-support).
+[OpenStack](https://specs.openstack.org/openstack/nova-specs/specs/mitaka/implemented/virt-driver-cpu-thread-pinning.html),
+which is one of the leading platform for VNF (Virtual Network Functions), the predecessor of CNFs.
+
+#### Improve the density of running containers
+
+Thread allocation guarantees enables more efficient usage of the node resources.
+The nodes can now accommodate safely mixed workloads of latency-sensitive and not-latency-sensitive (infrastructure) pods.
+This increases the node usage, which reduces the need for extra hardware, which drives down the TCO of a container-based solution.
+
+### Risks and Mitigations
+
+This new behaviour is opt-in. Users will need to explicitly enable it in their kubelet configuration. The change is very self contained, with little impact in the shared codebase.
+The impact in the shared codebase will be addressed enhancing the current testsuite.
+
+| Risk                                                      | Impact        | Mitigation |
+| --------------------------------------------------------- | ------------- | ---------- |
+| Bugs in the implementation lead to kubelet crash | High | Disable the policy and restart the kubelet. The workload will run but with weaker guarantees - like it was before this change. |
+
+
+## Design Details
+
+### Proposed Change
+
+We propose to
+- add a new _alias_ to the existing static policy, called `configurable`
+- add a new flag in Kubelet called `CPUManagerPolicyOptions` in the kubelet config or command line argument called `cpumanager-policy-options` which allows the user to specify the CPU Manager policy option.
+- finally, add a new cpu manager option called `reject-non-smt-aligned`; if present, this option will enable further refinements of the existing static policy.
+
+The addition of an alias for the `static` policy is motivated by the fact the new CPUManager options can alter the behaviour of the policy in a backward incompatible way.
```

Comment:
The `static` policy as understood today is a baseline.  I am not sure I understand the `configurable` alias as that alias has the same meaning as the existing `static` policy.  At the moment, I think its fine to just have the policy options bit because 'configurable' has no meaning yet until such time we enumerate each behavior as named policy tokens.

Reply:
basically, let's remove `configurable` alias at the moment until a future time lets you declare all attributes as policy options that inform a policy.